code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm

```diff
@@ -2,1474 +2,1217 @@
+	max_integrity = 500
+	soft_armor = list(MELEE = 60, BULLET = 50, LASER = 40, ENERGY = 40, BOMB = 0, BIO = 100, FIRE = 100, ACID = 0)
+	hit_sound = SFX_BEHEMOTH_EARTH_PILLAR_HIT
+	destroy_sound = 'sound/effects/alien/behemoth/earth_pillar_destroyed.ogg'
+	obj_flags = CAN_BE_HIT|BLOCKS_CONSTRUCTION
+	xeno_structure_flags = IGNORE_WEED_REMOVAL
+	allow_pass_flags = PASS_LOW_STRUCTURE|PASS_THROW|PASS_AIR|PASS_WALKOVER
+	hud_possible = HEALTH_HUD_XENO
+	/// List of connections used to check for climbing and other stuff.
+	var/static/list/connections = list(
+		COMSIG_OBJ_TRY_ALLOW_THROUGH = PROC_REF(can_climb_over),
+		COMSIG_FIND_FOOTSTEP_SOUND = TYPE_PROC_REF(/atom/movable, footstep_override),
+		COMSIG_TURF_CHECK_COVERED = TYPE_PROC_REF(/atom/movable, turf_cover_check),
+	)
+	/// The xeno owner that created this object. NOT the holder.
+	var/mob/living/carbon/xenomorph/xeno_owner
+	/// The atom that's currently holding this object.
+	var/mob/living/carbon/xenomorph/current_holder
+	/// References the dummy item created when this object is held.
+	var/obj/item/pillar_item/dummy_item
+	/// Visual overlay applied over this object. Mostly used for warning flashes.
+	var/obj/effect/pillar_flash/flash_visual
+	/// Cooldown for exploding so that we don't get infinite loops.
+	COOLDOWN_DECLARE(explosion_cooldown)
+
+/obj/structure/xeno/earth_pillar/Initialize(mapload, mob/living/carbon/xenomorph/new_owner, health_reduction)
 	. = ..()
-	switch(direction)
-		if(NORTH)
-			pixel_x += 12
-			pixel_y += 1
-		if(SOUTH)
-			pixel_x -= 10
-			pixel_y -= 1
-		if(WEST)
-			pixel_x -= 25
-			pixel_y -= 1
-		if(EAST)
-			pixel_x += 32
-			pixel_y -= 1
+	if(health_reduction)
+		max_integrity -= floor(max_integrity * health_reduction)
+		obj_integrity = max_integrity
+	transform = matrix(rand(1, 360), MATRIX_ROTATE) // Variety is the spice of life.
+	xeno_owner = new_owner
+	flash_visual = new
+	flash_visual.alpha = 0
+	flash_visual.layer = layer + 0.01
+	vis_contents += flash_visual
+	AddElement(/datum/element/connect_loc, connections)
+	RegisterSignal(src, COMSIG_MOVABLE_PRE_THROW, PROC_REF(pre_throw))
+	prepare_huds()
+	for(var/datum/atom_hud/xeno/xeno_hud in GLOB.huds)
+		xeno_hud.add_to_hud(src)
+	update_visuals()
+	RegisterSignals(src, list(COMSIG_ATOM_BULLET_ACT, COMSIG_ATOM_EX_ACT, COMSIG_ATOM_ATTACK_HAND, COMSIG_ATOM_ATTACK_HAND_ALTERNATE, COMSIG_ATOM_ATTACKBY), PROC_REF(update_visuals))
+
+/obj/structure/xeno/earth_pillar/Destroy()
+	vis_contents -= flash_visual
+	qdel(flash_visual)
+	playsound(loc, 'sound/effects/alien/behemoth/earth_pillar_destroyed.ogg', 40, TRUE)
+	new /obj/effect/temp_visual/behemoth/earth_pillar/creation/destruction(loc)
+	if(xeno_owner)
+		SEND_SIGNAL(src, COMSIG_EARTH_PILLAR_DESTROY) // Surprised Destroy() doesn't send a signal, but hey, we make do.
+		xeno_owner = null
+	if(current_holder)
+		force_drop()
+	return ..()
 
-/obj/effect/temp_visual/behemoth/stomp/east/Initialize(mapload, direction)
+/* to do: need damage icons
```

Comment:
hi todo

Reply:
Yeah, I do need sprites for Earth Pillar. I'll see if I can get those sometime soon.